include/boost/url/impl/const_string.ipp

```diff
@@ -128,69 +128,67 @@ const_string::
 const_string(
     result const& r) noexcept
     : string_view(
-        r.data, r.size)
-    , p_(r.p)
+        r.data, r.size),
+        data_(r.p)
 {
 }
 
 const_string::
 ~const_string()
 {
-    p_->release(size());
+    if (size() > sizeof(base*)) {
+        data_.p_->release(size());
+    }
 }
 
 const_string::
 const_string() noexcept
-    : string_view()
-    , p_([]
-    {
-        struct empty_string final : base
-        {
-            void
-            destroy(std::size_t) noexcept override
-            {
-            }
-        };
-
-        struct deleter
-        {
-            empty_string* p;
-
-            ~deleter()
-            {
-                p->release(0);
-            }
-        };
-
-        static empty_string cs{};
-        static deleter d{&cs};
-
-        ++cs.refs;
-
-        return &cs;
-    }())
+    : string_view(data_.buf_, 0)
 {
 }
 
 const_string::
 const_string(
     const_string const& other) noexcept
     : string_view(other)
```

Comment:
if you are going to completely initialize the object in the function body, then you don't need to delegate the construction. Also don't you need to call `p_->release()` in the else body?

Reply:
> if you are going to completely initialize the object in the function body, then you don't need to delegate the construction. 

Oh... right. It's because I was only initializing in the function body when the condition was true. But that wasn't a good idea.

> Also don't you need to call p_->release() in the else body?

In the else body, there's nothing to release from other because other is using SSO. And there's nothing to release from this because it's still being constructed.

Reply:
Ah right. it could use a comment :) do be aware of the rules for object destruction when a constructor throws. If the delegated constructor succeeds, and the body of the delegating constructor throws, then the object will have the destructor called.

Reply:
Actually, I have to do both:

```
 warning: base class ‘class boost::core::basic_string_view<char>’ should be explicitly initialized in the copy constructor [-Wextra]
```

I'm including the comment.

> a constructor throws

I'm using a noexcept constructor in the initializer. But the constructor overload we are using for `string_view(other.data(), other.size())` is not noexcept. Should we remove noexcept from the copy constructor?

Reply:
Same as another comment below